src/feature/hs/hs_circuit.c

```diff
@@ -1130,14 +1144,14 @@ hs_circ_send_introduce1(origin_circuit_t *intro_circ,
              "Failing.", TO_CIRCUIT(intro_circ)->n_circ_id);
     goto done;
   }
-  setup_introduce1_data(ip, exit_node, subcredential, &intro1_data);
-  /* If we didn't get any link specifiers, it's because our node was
-   * bad. */
-  if (BUG(!intro1_data.link_specifiers) ||
-      !smartlist_len(intro1_data.link_specifiers)) {
-    log_warn(LD_REND, "Unable to get link specifiers for INTRODUCE1 cell on "
-             "circuit %u.", TO_CIRCUIT(intro_circ)->n_circ_id);
-    goto done;
+
+  /* We should never select an invalid rendezvous point in theory but if we
+   * do, this function will fail to populate the introduce data. */
+  if (BUG(setup_introduce1_data(ip, exit_node, subcredential,
```

Comment:
Hm, why are we wrapping this into a BUG()? If it's a BUG, why do we introduce whole new behavior just for a buggy possibility? Instead we should fix the bug.

Or is this an edge-case that can happen normally? In this case we should not call BUG.